src/mcmcchains-connect.jl

```diff
@@ -27,4 +27,78 @@ function AbstractMCMC.bundle_samples(
 
     # Bundle everything up and return a Chains struct.
     return Chains(vals, param_names, (internals=["lp"],))
+end
+
+# function AbstractMCMC.bundle_samples(
+#     rng::Random.AbstractRNG, 
+#     model::DensityModel, 
+#     s::Ensemble, 
+#     N::Integer, 
+#     ts::Vector{<:Walker},
+#     chain_type::Type{Chains}; 
+#     param_names=missing,
+#     kwargs...
+# )
+#     # return ts
+#     vals = mapreduce(
+#         t -> map(i -> vcat(ts[t].walkers[i].params, 
+#                  ts[t].walkers[i].lp, t, i),
+#                  1:length(ts[t].walkers)), 
+#         vcat, 
+#         1:length(ts))
+    
+#     vals = Array(reduce(hcat, vals)')
+
+#     # return vals
+
+#     # Check if we received any parameter names.
+#     if ismissing(param_names)
+#         param_names = ["param_$i" for i in 1:length(ts[1].walkers[1].params)]
+#     else
+#         # Deepcopy to be thread safe.
+#         param_names = deepcopy(param_names)
+#     end
+
+#     # Add the log density field to the parameter names.
+#     push!(param_names, "lp", "iteration", "walker")
+
+#     # Bundle everything up and return a Chains struct.
+#     return Chains(vals, param_names, (internals=["lp", "iteration", "walker"],))
+# end
+
+function AbstractMCMC.bundle_samples(
+    rng::Random.AbstractRNG, 
+    model::DensityModel, 
+    s::Ensemble, 
+    N::Integer, 
+    ts::Vector,
+    chain_type::Type{Chains}; 
+    param_names=missing,
+    kwargs...
+)
+    # return ts
+    vals = mapreduce(
+        t -> map(i -> vcat(ts[t][i].params, 
+                 ts[t][i].lp, t, i),
```

Comment:
I think it is a bit strange to include the iteration here as an additional value since it is inconsistent with the design and all other uses of Chains so far. By design, the iteration corresponds to the first dimension. Maybe it would be better to return either a `Chains` object for every walker or to have rename the parameters and log probabilities for walker `i` to `param_j_i` and `lp_i`.

Reply:
That would make the chain essentially useless, since ensemble sampling suggests large numbers of walkers (> 500-1000). You'd have an astronomical number of parameters.

Getting it into chain form is mostly just a quick way to get summary stats on your posterior. If you really want to do walker analysis, it'll take us a good deal longer to find an adequate data structure other than the raw set of walkers. If I had more time and inclination and we weren't so restrictive on how AxisArrays were used, I would just add another axis for each walker.

Plus, lumping them all in together gets us damn close to MH:

EMCEE:
```julia
Summary Statistics
  parameters     mean     std  naive_se    mcse        ess   r_hat
  ──────────  ───────  ──────  ────────  ──────  ─────────  ──────
       sigma  -0.0681  0.0175    0.0004  0.0005  2494.1845  0.9996
          mu   1.0281  0.0118    0.0002  0.0003  1655.8455  1.0044
```

MH:

```julia
Summary Statistics
  parameters     mean     std  naive_se    mcse     ess   r_hat
  ──────────  ───────  ──────  ────────  ──────  ──────  ──────
       sigma  -0.0705  0.0055    0.0002  0.0013  6.3517  1.0591
          mu   1.0130  0.0120    0.0005  0.0029  6.3517  1.0591
```

Reply:
Ah the stats are nice! Yeah, I'm not sure what's the most reasonable way if one wants to throw the results into a Chains object. It just feels like a severe misuse of the design of Chains if one adds an `iteration` parameter. Maybe the easiest solution (still a misuse of design though) would be to concatenate the walkers along the last dimension, i.e., misuse the dimension for chains to indicate the walkers. Then all analysis should work as expected, the iteration number in the first dimension would actually correspond to the actual iteration, and the parameters would be as expected, and there would be no need for an additional `walker` and `iteration` parameter. What do you think?

Reply:
That's not a bad idea, I can do that.

Reply:
Added.